### PR TITLE
fixed #9595 - Primitve data not collapsible

### LINKED ISSUE #9595: a11y:  Once you expand a data element of a schema, there is no way to collapse it again.
### Q&A (please complete the following information)
 - OS: [e.g. macOS]  Windows 10
 - Browser: [e.g. chrome, safari]  Chrome
 - Version: [e.g. 22]  121
 - Method of installation: [e.g. npm, dist assets]  npm
 - Swagger-UI version: [e.g. 3.10.0]  Swagger UI React 5.11.3
 - Swagger/OpenAPI version: [e.g. Swagger 2.0, OpenAPI 3.0]  OpenAPI 3.0

### Content & configuration
Basic/ default

### Describe the bug you're encountering
Once you expand a data element of a schema, there is no way to collapse it again.
It looks like collapsed data elements have a "td" with buttons containing class "model-box-control and span with classes "model-toggle collapsed" which are NOT included for the elements that are already expanded.

### To reproduce...
Expand Schema, expand an input or output schema, expand a data element.  Once expanded, there is no way to contract it again.

### Expected behavior
Once expanded, it should be collapsible

### Screenshots
Note for the expanded elements, there is no caret to collapse it again.
image

### PR DESCRIPTION
**Pull request by dakkkshh**
Primitve data not collapsible. Changed hideSelfOnExpand prop to false for primitive-model

### Description
Primitive data can be expaned but can not be collapsed. This is due to the prop hideSelfOnExpand which used to check the depth of data. I set it to false. Now whether the data is primitve / not, it can be collapsed.

### Motivation and Context
"Fixes #9595 "

### How Has This Been Tested?
I manually tested the fix by launching the application and checking whether primitive data can be collapsed or not. I also rna the test script and here is the result:
Total Tests: 462
Passed: 459
Failed: 0
Pending: 3

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/core/components/primitive-model.jsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default class Primitive extends Component {
       </span>
 
     return <span className="model">
-      <ModelCollapse title={titleEl} expanded={depth <= expandDepth} collapsedContent="[...]" hideSelfOnExpand={expandDepth !== depth}>
+      <ModelCollapse title={titleEl} expanded={depth <= expandDepth} collapsedContent="[...]" hideSelfOnExpand={false}>
```
Comment (glowcloud) [inline src/core/components/primitive-model.jsx:53]: `hideSelfOnExpand` can be removed, as it's `false` by default in the `ModelCollapse` component:
```js
<ModelCollapse title={titleEl} expanded={depth <= expandDepth} collapsedContent="[...]"
```

**Review by glowcloud — CHANGES_REQUESTED** [review]
Hi  @dakkkshh,

Thanks for this fix! Please take a look at my comment to see the suggested change.

The PrimitiveModel component was also moved to the `json-schema-5` plugin in https://github.com/swagger-api/swagger-ui/commit/3b72ee18bcfa7822ccf3b82666646836f897fdd8. Could you update your PR to include this change there?

File: src/core/components/primitive-model.jsx (hunk 1 above)
Comment (dakkkshh) [reply src/core/components/primitive-model.jsx]: Hey @glowcloud 

I have resolved the mentioned change. Please have a look at the updated PR.

**Review by glowcloud — APPROVED** [review]

**Comment by glowcloud** [comment]
Thanks for the changes! LGTM!

**Pull request merged**